incubator/elasticsearch/templates/client-deployment.yaml

```diff
@@ -63,6 +63,10 @@ spec:
               fieldPath: metadata.namespace
         - name: NODE_DATA
           value: "false"
+{{- if hasPrefix "5." .Values.image.tag }}
```

Comment:
What if version 6 comes out or someone uses a custom image that has different versioning?

Reply:
@unguiculus: if version 6 comes out, we'll need to update the chart, but we'll probably need to update the chart for a major release anyway 😃. How would you prefer this to be configured? Would you rather the chart just support a single elastic version? Or do you think we should just special-case 2.x, like this:

```yaml
{{- if hasPrefix "2." .Values.image.tag }}
bootstrap.mlockall: ${BOOTSTRAP_MLOCKALL:false}
{{- else }}
bootstrap.memory_lock: ${BOOTSTRAP_MEMORY_LOCK:false}
{{- end }}
```

Reply:
Maybe you could add a separate value `esVersion` instead of relying on the image version. That would allow me to specify a custom image that is versioned differently.

Reply:
I went ahead and updated the chart with a values variable as suggested.

Looks like the concept of `appVersion` is already baked in to the `Chart.yaml`, unfortunately it is [immutable](https://github.com/kubernetes/helm/issues/2566#issuecomment-307964606).

more info:
- [The Chart.yaml File](https://github.com/kubernetes/helm/blob/1fa853959ccbdb1c2834bd625f3655b2a9e7a4cb/docs/charts.md#the-chartyaml-file)
- [The appVersion field](https://github.com/kubernetes/helm/blob/1fa853959ccbdb1c2834bd625f3655b2a9e7a4cb/docs/charts.md#the-appversion-field)